pkg/virt-controller/services/template.go

```diff
@@ -702,6 +703,16 @@ func (t *templateService) RenderLaunchManifest(vmi *v1.VirtualMachineInstance) (
 	}
 
 	nodeSelector[v1.NodeSchedulable] = "true"
+	if t.nodeSelector != "" {
+		for _, s := range strings.Split(t.nodeSelector, ",") {
+			v := strings.Split(strings.TrimSpace(s), "=")
+			if len(v) != 2 {
+				log.Log.Warningf("Invalid nodeSelector label: %s", v)
```

Comment:
Since is passed in as a cli arg, I think this validation should be occurring during virt-controller's initialization. We should be parsing the node selector arg earlier than this to ensure it's valid.  I'd rather see virt-contoller fail to start than for this to silently fail and let the VMI launch without the node selector.

Reply:
Parse once, and reuse.